### ISSUE TITLE
Allow insecure external registries

### ISSUE THREAD

**Issue by pedro-brito-ilia** [epinio/epinio#1871]
I'm iterating using JFrog Artifactory as an external registry for pushing and pulling images to Epinio. As of right now my deployment is local and the registry is deployed on the same cluster as Epinio, but even though all configurations are set to deploy my registry on insecure mode, Epinio apparently does not provide an option for this.

Is there a way to specify connection on port 80, instead of using HTTPS? The only variables Epinio helm chart provides for external registries are:
  - registryURL
  - registryUsername
  - registryPassword
  - registryNamespace

**Additional details**

- Registry is deployed below the same reverse proxy as Epinio (Traefik)
- Insecure registry variable is set on Docker daemon.json

**Logs**
🕞  [stage-workspace-conda--6aa3301d202ac865204bb6918ba9f6da1a7cjhdh] buildpack ERROR: failed to initialize analyzer: getting previous image: connect to repo store "artifactory.172.17.0.1.sslip.io/docker-local/workspace-conda-app:9c446d2d03b5ac20": Get "https://artifactory.172.17.0.1.sslip.io/v2/": x509: certificate is valid for 0114a4c3e0281be95e112c9f6eb62f2b.962a60ba4f0e9cf5a64a97c6fdecb69c.traefik.default, not artifactory.172.17.0.1.sslip.io

**Comment by pedro-brito-ilia** [comment]
More info: Using self-signed certificate also does not work.
🕞  [stage-workspace-conda--658d4f1c4012f208c6288a0737428f832efgg4t4] buildpack ERROR: failed to initialize analyzer: getting previous image: connect to repo store "artifactory.172.17.0.1.sslip.io/docker-local/workspace-conda-app:9c446d2d03b5ac20": Get "https://artifactory.172.17.0.1.sslip.io/v2/": x509: certificate signed by unknown authority

**Comment by enrichman** [comment]
Hi @pedro-brito-ilia, unfortunately it's not possible to use an insecure mode because of an open issue on the lifecycle builder:
- https://github.com/buildpacks/lifecycle/issues/524

You can find a more detailed [doc here](https://github.com/epinio/epinio/blob/main/docs/explanations/architecture/container-registry.md).

It should be possible to use a self-signed certificate, but you need to tell to the Epinio server about it.  
The `REGISTRY_CERTIFICATE_SECRET` env var is used to fetch and mount the certificate used during the build stage. This need to be in the same `epinio` namespace.

Can you provide more info about your deployment, and how to reproduce it?  
I'm not familiar with Artifactory, and if I can install it in the same way on my local env it would be easier to test.

**Comment by gitricko** [comment]
> Can you provide more info about your deployment, and how to reproduce it? I'm not familiar with Artifactory, and if I can install it in the same way on my local env it would be easier to test.

Hi @enrichman, here is some instruction how to install jfrog



```sh
helm repo add jfrog https://charts.jfrog.io
helm repo update
helm install artifactory --create-namespace -n jfrog jfrog/artifactory-jcr \
  --values arti-values.yaml
kubectl apply -f arti-cert.yaml
kubectl rollout status statefulset.apps/artifactory -n jfrog --timeout=480s
```

Sample arti-values and arti-cert.yaml

arti-cert.yaml
```
---
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  namespace: jfrog
  name: articert
spec:
  secretName: arti-tls
  issuerRef:
    kind: ClusterIssuer
    name: selfsigned-issuer
  commonName: 'arti.internal.192.168.69.126.sslip.io'
  dnsNames:
    - 'arti.internal.192.168.69.126.sslip.io'
```

arti-values.yaml
```
artifactory:
  nginx:
    enabled: true
    service:
      type: NodePort
  ingress:
    enabled: true
    hosts: ['arti.internal.192.168.69.126.sslip.io']
    tls:
      - hosts:
        - arti.internal.192.168.69.126.sslip.io
        secretName: arti-tls
  postgresql:
    enabled: false
  customCertificates:
    enabled: true
    certificateSecretName: arti-tls
```

Installing epinio:
```
		helm install epinio -n epinio --create-namespace epinio/epinio \
			--set global.domain=internal.192.168.69.126.sslip.io \
			--set global.tlsIssuer=selfsigned-issuer \
			--set containerregistry.enabled=false \
			--set global.registryURL=arti.internal.192.168.69.126.sslip.io \
			--set global.registryNamespace=docker-local \
			--set global.registryUsername=admin \
			--set global.registryPassword=Passwor1
```

**Comment by enrichman** [comment]
Thanks @gitricko, I tried a couple of things but that environment variable is set only with the `containerregistry.enabled`.

I tried a couple of things but I probably need more time to figure out how to set up everything correctly. By the way I've seen that last week the lifecycle issue was labeled as "ready" (https://github.com/buildpacks/lifecycle/issues/524#event-8550999601) and this will probably simplify a lot the internal registry deployment and setup.
